### ISSUE TITLE
Connection closed error on production in Vercel

### ISSUE THREAD

**Issue by vehktaur** [vercel/next.js#72407]
### Link to the code that reproduces this issue

https://github.com/vehktaur/blogger

### To Reproduce

Start the application in deployment @ blogger-vehktaur.vercel.app

Error occurs while trying to load the blogs on the home page.

### Current vs. Expected behavior

This behaviour is new, and everything worked as expected on deployment previously.

### Provide environment information

```bash
Operating System: Windows 11 Pro
```

### Which area(s) are affected? (Select all that apply)

create-next-app, Image (next/image), TypeScript

### Which stage(s) are affected? (Select all that apply)

Vercel (Deployed)

### Additional context

The app works completely fine locally but gives a connection closed error on deployment 
Basically, it's the exact same issue here #52824

**Comment by ztanner** [comment]
Hi,

It seems like your database call is timing out. On Vercel, Hobby accounts have a default limit of 10s to return a response. If they don't respond in time, the function exits early. You'll notice that the connection closed message consistently appears at the around the 10s mark.

You're able to configure this up to 60s: https://vercel.com/docs/functions/configuring-functions/duration

If you're still seeing this after bumping the timeout, you might need to investigate your database call to ensure it's completing successfully. Putting a log statement before & after the database call should be able to provide more information on timing. 

If you're still having issues getting it to work, please check https://vercel.com/help to get platform help.

**Comment by vehktaur** [comment]
it worked. thanks.

> Hi,
> 
> It seems like your database call is timing out. On Vercel, Hobby accounts have a default limit of 10s to return a response. If they don't respond in time, the function exits early. You'll notice that the connection closed message consistently appears at the around the 10s mark.
> 
> You're able to configure this up to 60s: https://vercel.com/docs/functions/configuring-functions/duration
> 
> If you're still seeing this after bumping the timeout, you might need to investigate your database call to ensure it's completing successfully. Putting a log statement before & after the database call should be able to provide more information on timing.
> 
> If you're still having issues getting it to work, please check https://vercel.com/help to get platform help.
